### ISSUE TITLE
Better search for Boost

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#183]
It is not uncommon to have different components of Boost to be installed in different directories (`boost-mpi` is the most common one). We need to document how to (easily!) explain ALPSCore that it should look for Boost components in different directories.

**Comment by aeantipov** [comment]
This one is hard to achieve. In fact the only working solution I found is to put libraries explicitly.
Some along these lines: 
```
CXX="mpic++ -L$TACC_BOOST_MPI_LIB -I$TACC_BOOST_MPI_INC" CC="mpicc" cmake \
-DBOOST_ROOT=$TACC_BOOST_DIR \
-DBOOST_LIBRARYDIR=$TACC_BOOST_LIB \
-DBoost_MPI_LIBRARY_RELEASE=${TACC_BOOST_MPI_LIB}/libboost_mpi.a \
-DBoost_SYSTEM_LIBRARY_RELEASE=${TACC_BOOST_LIB}/libboost_system.a \
...
```

**Comment by egull** [comment]
If we remove boost MPI (which is often separate and in a separate directory) I think we can automatically find it in most cases – and then manually (a la Andrey) deal with the rest of cases. The main difficulty here is the MPI library which has to be compiled against the MPI, compiler version, and boost version that is used for a release. The libraries themselves are much more benign.

**Comment by aeantipov** [comment]
It seems this one is closed with 0.5.3 release
